doc/languages-frameworks/agda.section.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Agda
+author: Alex Rice (alexarice)
+date: 2020-01-06
+---
+# Agda
+
+## How to use Agda
+
+Agda can be installed from `agda`:
+```
+$ nix-env -iA agda
+```
+
+To use agda with libraries, the `agda.withPackages` function can be used. This function either takes:
++ A list of packages,
++ or a function which returns a list of packages when given the `agdaPackages` attribute set,
++ or an attribute set containing a list of packages and a GHC derivation for compilation (see below).
+
+For example, suppose we wanted a version of agda which has access to the standard library. This can be obtained with the expressions:
+
+```
+agda.withPackages [ agdaPackages.standard-library ]
+```
+
+or
+
+```
+agda.withPackages (p: [ p.standard-library ])
+```
+
```

Comment:
```suggestion

or

\```
agda.withPackages {
  pkgs = [ agdaPackages.standard-library ];
}
\```
```

Reply:
I don't think this needs to be changed

Reply:
> I don't think this needs to be changed

I agree. I think non-agda dependencies don't belong to the `withPackages` interface.

Reply:
More just looking at the file it looks like this changes it to say the same thing twice but maybe I am not understanding the diff

Reply:
@alexarice 
No, it's not the same. It's meant to cover case when:
```
agda.withPackages {
  pkgs = [ agdaPackages.standard-library ];
  ghc = /* ... */;
}
```
but again, this interface should not exist in the first place IMO.

Reply:
Ah I see, this is just adding an extra thing. I still don't think it is necessary as it occurs below as well. I'm not sure what you mean about the interface should not exist?

Reply:
> it occurs below as well

What I wrote (with `ghc` argument) occurs below. But the suggested option without ghc is just longer, and doesn't add anything.

> I'm not sure what you mean about the interface should not exist?

I was thinking that
```
runCommandNoCC "${pname}-${version}" {
    inherit ghc;
  } ''
    "$ghc"/bin/ghc --version
  '';
```
would be overridable as `(agda.withPackages (p: [p.standard-library])).override { ghc = haskell.compiler.ghcHEAD; }`.

This, of course, depends on how often you need to override `ghc`.  https://github.com/NixOS/nixpkgs/pull/76653#issuecomment-576810802 mentions Haskell FFI, so, perhaps, this may be a needed option.

Reply:
My reasoning to add this is to erradicate the difference in the docs where it first says "there are three ways to do libraries", and then only two examples are shown.

Reply:
The solution here seems to be changing the 3 to a 2

Reply:
ah nevermind I see what you mean now